[DO NOT MERGE] EQ IIR conversion Take 2

ranj063 opened the pull request.

ranj063 committed:
eq_iir: Use the new module interface

Use the module interface instead of the comp drv interface for the
EQ IIR component.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

ranj063 committed:
audio_format: Add macro for converting bytes <-> samples

And macros for converting bytes to samples and samples to bytes and use
it in all components to avoid duplication.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

ranj063 committed:
module_adapter: Add a new helper to update input/output stream buffers

Add a new helper to update the consumed bytes in the input stream
buffer and the produces bytes in the output stream buffer.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

ranj063 committed:
module_adapter: set state to INITIALIZED after reset

A reset will always be followed by a prepare. So set the state to
MODULE_INITIALIZED after reset so that the following prepare() can
proceed without issues.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

fredoh9 commented:
This is passed with check-signal-stop-start.sh on ADLP_RVP_NOCODEC. run-test-all.sh -l 1 is running now, looking good so far.

fredoh9 commented:
Failed in multiple-pipeline-capture.sh, DSP panic. We can wait and see official PR testing result as well.

```
+ /home/ubuntu/sof-test/test-case/multiple-pipeline-capture.sh -l 1
2022-07-19 20:47:24 UTC Sub-Test: [INFO] DUT is running IPC3 mode
...
===========================>>
[  974.550619] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ DSP dump start ]------------
[  974.550620] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: DSP panic!
[  974.550620] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: fw_state: SOF_FW_BOOT_COMPLETE (6)
[  974.550625] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: 0x00000005: module: ROM, state: FW_ENTERED, running
[  974.550627] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: status code: 0xdead006 (unknown)
[  974.550713] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: reason: runtime exception (0x6)
[  974.550714] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: trace point: 0x00004000
[  974.550715] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: panic at :0
[  974.550715] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: DSP Firmware Oops
[  974.550716] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: Exception Cause: LoadStorePIFDataErrorCause, Synchronous PIF data error during LoadStore access
```

aiChaoSONG commented:
part of this pr is already merged，only two commits valid
- module_adapter: set state to INITIALIZED after reset
- data_blob: modify comp_data_blob_handler_free()

lyakh commented on src/audio/data_blob.c:
```diff
@@ -476,10 +476,8 @@ comp_data_blob_handler_new_ext(struct comp_dev *dev, bool single_blob,
-	rfree(blob_handler);
+	if (blob_handler) {
+		comp_free_data_blob(blob_handler);
+		rfree(blob_handler);
```
this commit looks like a NOP to me and it doesn't match the commit description

ranj063 commented:
> part of this pr is already merged，only two commits valid
> 
> * module_adapter: set state to INITIALIZED after reset
> * data_blob: modify comp_data_blob_handler_free()

@aiChaoSONG i'm aware of that and thats why it is still a draft. I will make it ready for review when I have clarity

ranj063 committed:
edf_schedule: Only set the interrupt if the CPU is not powered down

Prevent the xtos EDF scheduler from setting the interrupt when the CPU
power down has already started. This fixes an IDC timeout when powering
down the secondary cores with xtos.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

lgirdwood commented:
> +bool cpu_power_down[CONFIG_CORE_COUNT];
> +

Not cache coherent, Any array of elems where the elems are not multiples of cache line size are incoherent.

ranj063 commented:
@lgirdwood @lyakh I've updated the PR with Guennadi's suggested patch. Please let me know if it looks ok

lyakh commented:
> > +bool cpu_power_down[CONFIG_CORE_COUNT];
> > +
> 
> Not cache coherent, Any array of elems where the elems are not multiples of cache line size are incoherent.

@lgirdwood we added `SHARED_DATA` to the array definition, it should be cache-coherent now. Also notice, that this change is XTOS-only, these files aren't used with Zephyr, and we already use static shared arrays of per-core data in other locations, e.g.
```
src/lib/notifier.c:static SHARED_DATA struct notify_data notify_data_shared[CONFIG_CORE_COUNT];
src/idc/idc.c:static SHARED_DATA struct idc_payload static_payload[CONFIG_CORE_COUNT];
```

singalsu commented:
@ranj063 This seems to be an abandoned PR. The other PR that I used for new #6992 was earlier with name "Take 3" before rename to current.

sys-pt1s commented:
Can one of the admins verify this patch?
